### PR TITLE
Summary Search API

### PR DESCRIPTION
**Pull request by profjsb**
Facilitate the search for sources using natural language. This PR makes use of the summary embeddings, leveraging `langchain` to identify the top k sources that match a query.  From the CLI:

```bash
curl -s -X GET -H 'Authorization: token <user_token>' http://...:5000/api/summary_query \
     -d '{"q":"What sources are associated with NGC galaxies?", "k": 1, "z_min": 0.004}'
```
returns something like:
```json
{
  "status": "success",
  "data": {
    "query_results": [
      {
        "id": "ZTF23.....",
        "score": 0.830897391,
        "metadata": {
          "class": [
            "Type II"
          ],
          "redshift": 0.002,
          "summary": "Based on the information provided, it seems that a Type II supernova (SN II) has been observed in NGC XXX with a redshift of 0.005. The supernova peaked fainter than -15 and has an absolute magnitude of -14.5. There are also HST data and STIS long slit spectra covering the SN position,...."
        }
      }
    ]
  },
  "version": "..."
```

### CODE REVIEW

File: services/openai_analysis_service/app.py [hunk 1]
```diff
@@ -273,11 +273,14 @@ def run_openai_summarization(data_dict):
         )
         pinecone_index = pinecone.Index(summarize_embedding_index)
         metadata = {}
-        if z is not None:
-            metadata["redshift"] = z
+        metadata["redshift"] = z if z is not None else -1
```
Comment (mcoughlin) [inline services/openai_analysis_service/app.py:276]: Is this strictly necessary? Should it only be used in certain circumstances?

File: skyportal/handlers/api/query.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
+import os
```
Comment (mcoughlin) [inline skyportal/handlers/api/query.py:1]: Maybe a more descriptive name for the .py file?

File: skyportal/handlers/api/query.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+setattr(Pinecone, 'search_sources', search_sources)
+
+# Preamble: get the embeddings and summary parameters ready
+# for now, we only support pinecone embeddings
+summarize_embedding_config = cfg[
+    'analysis_services.openai_analysis_service.embeddings_store.summary'
+]
+USE_PINECONE = False
+if (
```
Comment (mcoughlin) [inline skyportal/handlers/api/query.py:73]: Maybe we log if something is missing to help admins?

**Review by mcoughlin — APPROVED** [review]
Very cool. A few small recommendations.

File: services/openai_analysis_service/app.py (hunk 1 above)
Comment (profjsb) [reply services/openai_analysis_service/app.py:276]: I always want to have redshift in the metadata but might be able to get away without it.

Comment (mcoughlin) [reply services/openai_analysis_service/app.py:276]: I guess my worry is whether there will be confusion about a negative default redshift.

File: skyportal/handlers/api/query.py (hunk 3 above)
Comment (profjsb) [reply skyportal/handlers/api/query.py:73]: Done.

**Pull request merged**